### PR TITLE
feat: add sql functions uidStrToInt and uidIntToStr

### PR DESCRIPTION
**Pull request by ssoroka**
## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #

### CODE REVIEW

File: internal/server/data/schema/parse.go [hunk 1]
```diff
@@ -52,14 +58,14 @@ func ParseSchema(schema string) ([]Statement, error) {
...
-
 			var err error
 			if currentTable, line, err = parseStatementLine(line); err != nil {
 				return nil, err
 			}
 
 			currentStmt.WriteString("\n" + line + "\n")
-			if strings.HasSuffix(line, ";") {
+
+			if !inCodeBlock && strings.HasSuffix(line, ";") {
```
Comment (pdevine) [inline internal/server/data/schema/parse.go]: If you got to the scanSchemaForStartOfStatement an `inCodeBlock` was set true, we should probably panic.

**Review by dnephin — COMMENTED** [review]
Please include some detail in the PR description about how this will be used and what problem it solves

File: internal/server/data/sqlfunc.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				IF LENGTH(encoded) > 11 THEN
+					RAISE EXCEPTION 'invalid base58: too long';
+				END IF;
+
+				WHILE (i <= LENGTH(encoded)) LOOP
+				  IF (result > MAX_SIGNED_INT/58) THEN 
+						RAISE EXCEPTION 'invalid base58: value too large: %', encoded;
+					END IF;
+					result = result * 58;
+					pos := POSITION(substring(encoded FROM i FOR 1) in encodeBase58Map);
```
Comment (pdevine) [inline internal/server/data/sqlfunc.go]: I think you want SUBSTRING() here as well.

File: internal/server/data/sqlfunc_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+func TestSQLUidStrToIntRoundTrip(t *testing.T) {
+	type testCase struct {
+		base58 string
+		intval int64
+		err    string
+	}
+	db := setupDB(t, postgresDriver(t))
+	// err := sqlFunctionsMigration().Migrate(db)
```
Comment (pdevine) [inline internal/server/data/sqlfunc_test.go:17]: left over testing stuff?

File: internal/server/data/schema/parse.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ func ParseSchema(schema string) ([]Statement, error) {
 	var currentTable string
 	var currentStmt strings.Builder
 	var stmts []Statement
+	var inCodeBlock bool
```
Comment (dnephin) [inline internal/server/data/schema/parse.go:26]: Shouldn't `inCodeBlock` be part of the `state` switch, instead of a separate sub-state

**Review by pdevine — APPROVED** [review]
LGTM.

**Review by pdevine — APPROVED** [review]

**Pull request merged**